ros_buildfarm/templates/devel/devel_job.xml.em

```diff
@@ -245,6 +256,106 @@ if pull_request:
     'publisher_xunit',
     pattern='catkin_workspace/test_results/**/*.xml',
 ))@
+@[if not pull_request]@
+@(SNIPPET(
+    'publisher_groovy-postbuild',
+    script='\n'.join([
+        '// COLLATE BUILD TEST RESULTS AND EXPORT BUILD HISTORY FOR WIKI',
+        'import jenkins.model.Jenkins',
+        'import hudson.FilePath',
+        '',
+        '@Grab(\'org.yaml:snakeyaml:1.17\')',
+        'import org.yaml.snakeyaml.Yaml',
+        'import org.yaml.snakeyaml.DumperOptions',
+        '',
+        'manager.listener.logger.println("# BEGIN SECTION: Collate test results for wiki.")',
+        '',
+        '// nr of builds to include in history',
+        'final num_build_hist = 5',
+        '',
+        'try {',
+        '  def data = [',
+        '    "history" : []',
+        '  ]',
+        '',
+        '  // gather info on tests of current build',
+        '  def tresult = manager.build.getAction(hudson.tasks.junit.TestResultAction.class)?.result',
+        '  if (tresult)',
+        '  {',
+        '    data.latest_build = [',
```

Comment:
Do we need to store `latest_build` or is the same information also available under the highest build number?

Reply:
An earlier version of this script did just refer to the highest build id in the `history` list.

But the history is collected later in the script, and could fail. If it does, we would not have any test data, not even for the current job. The test results of the current job are retrieved directly via the `TestResultAction`.

This is just a choice though: if we feel failure is failure (and we don't want to separate these two things) then having `latest_build` point to `data.history[0]` (or even removing `latest_build` entirely) is an option.

Reply:
Sounds good to me.